pkg/core/resources/apis/meshidentity/status/updater.go

```diff
@@ -148,10 +143,101 @@ func (i *IdentityProviderReconciler) initialize(ctx context.Context, mid *meshid
 			Message: "Provider successfully initialized",
 		})
 	}
+	if mid.Spec.Provider.Type == meshidentity_api.BundledType &&
+		pointer.DerefOr(mid.Spec.Provider.Bundled.MeshTrustCreation, meshidentity_api.MeshTrustCreationEnabled) == meshidentity_api.MeshTrustCreationEnabled {
+		if err := i.createOrUpdateMeshTrust(ctx, mid); err != nil {
+			conditions = append(conditions, common_api.Condition{
+				Type:    meshidentity_api.MeshTrustConditionType,
+				Status:  kube_meta.ConditionFalse,
+				Reason:  "MeshTrustCreationError",
+				Message: err.Error(),
+			})
+			return conditions
+		} else {
+			conditions = append(conditions, common_api.Condition{
+				Type:    meshidentity_api.MeshTrustConditionType,
+				Status:  kube_meta.ConditionTrue,
+				Reason:  "MeshTrustCreated",
+				Message: "MeshTrust has been successfully created",
+			})
+		}
+	}
 
 	return conditions
 }
 
+func (i *IdentityProviderReconciler) createOrUpdateMeshTrust(ctx context.Context, identity *meshidentity_api.MeshIdentityResource) error {
+	meshTrust := meshtrust_api.NewMeshTrustResource()
+	meshName := identity.Meta.GetMesh()
+	resourceName := identity.Meta.GetName()
+
+	trustDomain, err := identity.Spec.GetTrustDomain(identity.GetMeta(), i.zone)
+	if err != nil {
+		return err
+	}
+	// Check if the MeshTrust resource already exists
+	update := true
+	if err := i.roResManager.Get(ctx, meshTrust, store.GetByKey(resourceName, meshName)); err != nil {
+		if store.IsNotFound(err) {
+			update = false
+		} else {
+			return err
+		}
+	}
+	ca, err := i.loadCA(ctx, identity)
+	if err != nil {
+		return err
+	}
+
+	origin := kri.From(identity).String()
+	caPEM := string(ca)
+
+	if update {
+		// Check if the CA PEM is already present in the MeshTrust resource
+		for _, bundle := range meshTrust.Spec.CABundles {
+			if bundle.PEM.Value == caPEM {
```

Comment:
we only have one type right now but in the future there might be more, should we check the type here?

Reply:
is string comparison completely safe here? is there any chance that there is some additional whitespace char here and this equality could fail? should we `pem.Decode` it?

Reply:
Yes, should be better, changed